pkg/pgclient/client.go

```diff
@@ -114,6 +115,50 @@ func (cfg *Config) GetConnectionStr() string {
 		cfg.host, cfg.port, cfg.user, cfg.database, cfg.password, cfg.sslMode)
 }
 
+func (cfg *Config) GetNumConnections() (min int, max int, perProc int, err error) {
+	maxProcs := runtime.GOMAXPROCS(-1)
+	if maxProcs <= 0 {
+		maxProcs = runtime.NumCPU()
+	}
+	if maxProcs <= 0 {
+		maxProcs = 1
+	}
+	if cfg.ConnectionsPerProc != -1 {
+		return maxProcs, maxProcs * cfg.ConnectionsPerProc, cfg.ConnectionsPerProc, nil
+	}
+	perProc = 5
+	conn, err := pgx.Connect(context.Background(), cfg.GetConnectionStr())
+	if err != nil {
+		return 0, 0, 0, err
+	}
+	defer func() { _ = conn.Close(context.Background()) }()
+	row := conn.QueryRow(context.Background(), "SHOW max_connections")
+	var dbMaxConnections int
+	err = row.Scan(dbMaxConnections)
+	if err != nil {
+		return 0, 0, 0, err
+	}
+	if dbMaxConnections <= 1 {
+		log.Warn("msg", "database can only handle 1 connection")
+		return dbMaxConnections, dbMaxConnections, 1, nil
+	}
+	min = maxProcs
+	if dbMaxConnections <= min {
```

Comment:
I dont' know that half the connections is right, I'd say 80% is probably closer as a sort of failsafe here.